### ISSUE TITLE
XGBOOST regression models return reg:linear deprecation warning

### ISSUE THREAD

**Issue by aadler** [topepo/caret#1160]
## Issue
The `xgboost` package no longer uses `reg:linear` for its regression objective function but `reg:squarederror`. Running an xgboost model using `caret` will return meny errors of the form:

```r
[19:43:47] WARNING: amalgamation/../src/objective/regression_obj.cu:170: reg:linear is now deprecated in favor of reg:squarederror.
```
As these errors come from xgboost, they are not hidden by `warning=FALSE` chunk headers or even wrapping in `suppresWarnings`.  Looking at `xgbFit$finalModel$call` one sees `objective = "reg:linear"`. If this could be changed to `"reg:squarederror"`, that *should* solve the problem.

Note: My R installation is actually 4.0.2 on windows built from source using Rtools40, so that it why it appears as `r-devel`. 

Thank you.

### Minimal dataset:
```r
library(caret)
set.seed(2)
trainX <- matrix(runif(20, 1, 100), ncol = 2)
colnames(trainX) <- c('X1', 'X2')
trainY <- runif(10, 1, 20)
```
### Minimal, runnable code:
``` r
xgbFit <- train(x = trainX, y = trainY, method = 'xgbTree', tuneLength = 1L)
```
### Session Info:
```r
R Under development (unstable) (2020-06-21 r78727)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18362)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] caret_6.0-86    ggplot2_3.3.2   lattice_0.20-41

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5           pillar_1.4.4         compiler_4.1.0       gower_0.2.2          plyr_1.8.6           iterators_1.0.12     class_7.3-17         tools_4.1.0         
 [9] rpart_4.1-15         packrat_0.5.0        ipred_0.9-9          lubridate_1.7.9      lifecycle_0.2.0      tibble_3.0.1         gtable_0.3.0         nlme_3.1-148        
[17] pkgconfig_2.0.3      rlang_0.4.6          Matrix_1.2-18        foreach_1.5.0        rstudioapi_0.11      prodlim_2019.11.13   stringr_1.4.0        withr_2.2.0         
[25] dplyr_1.0.0          pROC_1.16.2          generics_0.0.2       vctrs_0.3.1          recipes_0.1.13       xgboost_1.1.1.1      stats4_4.1.0         nnet_7.3-14         
[33] grid_4.1.0           tidyselect_1.1.0     data.table_1.12.8    glue_1.4.1           R6_2.4.1             survival_3.2-3       lava_1.6.7           reshape2_1.4.4      
[41] purrr_0.3.4          magrittr_1.5         ModelMetrics_1.2.2.2 scales_1.1.1         codetools_0.2-16     ellipsis_0.3.1       MASS_7.3-51.6        splines_4.1.0       
[49] timeDate_3043.102    colorspace_1.4-1     stringi_1.4.6        munsell_0.5.0        crayon_1.3.4 
```

**Comment by leedrake5** [comment]
The default in caret should be changed, but you can do this directly through the train function:

`objective = "reg:squarederror"`

**Comment by arunrajes** [comment]
Thanks for clarifying.....
model6<-train(Y~.,
              data=df_train,
              objective="reg:squarederror",
              method='xgbTree',
              trControl=fitcontrol,
              verbose=FALSE)

**Comment by topepo** [comment]
closed via #1188
